### DISCUSSION TITLE
check access level of dir right after 'Select Destination Location' in installer

### DISCUSSION THREAD

**Discussion by goyalyashpal** [git-for-windows/git#4840] [Ideas]
### summaries

check access level of dir right after 'Select Destination Location' in installer

### steps

* run gfw setup "`Git-<ver>-<word-len>.exe`" as a user
* the next-next screens are as follows:
* select a protected dir in the "select destination location" screen
* on installation, the installation failed

```
Information (licence)
Select Destination Location
    dialog: "Folder Exists"
...
Installing
    dialog: "Git <ver> Setup" Setup was unable to create the directory "...". \n\nError 5: Access id denied"
```

### details

all the options picked/specified in all those screens are lost and have to be picked again while running the setup as admin.

so, the setup should immediately check for access level, and prompt for password immediately if access is not available rather than only at the very last step.

### duplicates

* this might be anti-parallel to the: https://github.com/git-for-windows/git/issues/4758

**Comment by dscho** [comment]
We use InnoSetup. It might not be possible to intercept the process at that level.

**Reply by goyalyashpal** [reply]
> intercept the process at that level.

does that mean _any_ logic based on user input, or just that _extreme_ kind of interruption (as in exit)?

if latter (just the extreme one) so, how about showing some notice of sorts in the installer itself?

**Reply by dscho** [reply]
I don't quite understand your question, but you can have a look at the source code of the installer: https://github.com/git-for-windows/build-extra/blob/main/installer/install.iss.
